packages/ui/client/composables/explorer/tree.ts

```diff
@@ -53,7 +54,8 @@ export class ExplorerTree {
     this.rafCollector = useRafFn(this.runCollect.bind(this), { fpsLimit: 10, immediate: false })
   }
 
-  loadFiles(remoteFiles: File[]) {
+  loadFiles(remoteFiles: File[], projects: string[]) {
+    this.projects.splice(0, this.projects.length, ...projects)
```

Comment:
this will require some changes when exposing the data for the filter selector, should be fixed in #7201 once merged